mountpoint-s3/src/data_cache/disk_data_cache.rs

```diff
@@ -0,0 +1,431 @@
+//! Module for the on-disk data cache implementation.
+
+use std::fs;
+use std::io::{ErrorKind, Read, Write};
+use std::ops::RangeBounds;
+use std::path::PathBuf;
+
+use bytes::Bytes;
+use mountpoint_s3_crt::checksums::crc32c::{self, Crc32c};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use tracing::{error, trace};
+
+use crate::data_cache::DataCacheError;
+
+use super::{BlockIndex, CacheKey, ChecksummedBytes, DataCache, DataCacheResult};
+
+/// Disk and file-layout versioning.
+const CACHE_VERSION: &str = "V1";
+
+/// On-disk implementation of [DataCache].
+pub struct DiskDataCache {
+    block_size: u64,
+    cache_directory: PathBuf,
+}
+
+/// Describes additional information about the data stored in the block.
+///
+/// It should be written alongside the block's data
+/// and used to verify it contains the correct contents to avoid blocks being mixed up.
+///
+/// TODO: Add checksum over struct to verify block metadata later.
```

Comment:
Not sure I understand: isn't this already in `header_checksum`? And/or shouldn't we include the block checksum as well?

Reply:
Sorry, yes! Missed removing that todo.